setup.py

```diff
@@ -1,74 +1,67 @@
 #!/usr/bin/env python
-import os
 import sys
-import re
 
-# require python 3.7 or newer
 if sys.version_info < (3, 7):
     print("Error: dbt does not support this version of Python.")
     print("Please upgrade to Python 3.7 or higher.")
     sys.exit(1)
 
 
-# require version of setuptools that supports find_namespace_packages
-from setuptools import setup
-
 try:
     from setuptools import find_namespace_packages
 except ImportError:
-    # the user has a downlevel version of setuptools.
     print("Error: dbt requires setuptools v40.1.0 or higher.")
-    print('Please upgrade setuptools with "pip install --upgrade setuptools" ' "and try again")
+    print('Please upgrade setuptools with "pip install --upgrade setuptools" and try again')
     sys.exit(1)
 
 
-# pull long description from README
-this_directory = os.path.abspath(os.path.dirname(__file__))
-with open(os.path.join(this_directory, "README.md")) as f:
-    long_description = f.read()
+from pathlib import Path
+from setuptools import setup
+
+
+README = Path(__file__).parent / "README.md"
+VERSION = Path(__file__).parent / "dbt/adapters/redshift/__version__.py"
+
 
+def dbt_redshift_version() -> str:
```

Comment:
shouldn't these functions be marked private?

Reply:
Technically yes, their only applicable scope is `setup.py`. For some reason I don't apply that to `setup.py` because I never think of someone importing from there. But we should be consistent. I'll make the change.